src/bandit-evaluator.ts

```diff
@@ -23,6 +23,45 @@ export class BanditEvaluator {
   private readonly assignmentShards = BANDIT_ASSIGNMENT_SHARDS; // We just hard code this for now
   private readonly sharder: Sharder = new MD5Sharder();
 
+  public evaluateBestBandit(
+    flagKey: string,
+    subjectKey: string,
+    subjectAttributes: ContextAttributes,
+    actions: Record<string, ContextAttributes>,
+    banditModel: BanditModelData,
+  ): BanditEvaluation {
+    const actionScores: Record<string, number> = this.scoreActions(
+      subjectAttributes,
+      actions,
+      banditModel,
+    );
+
+    const actionWeights: Record<string, number> = this.weighActions(
+      actionScores,
+      banditModel.gamma,
+      banditModel.actionProbabilityFloor,
+    );
+
+    const selectedActionKey = Object.keys(actionScores).reduce((prev, cur) => {
+      if (!prev || actionScores[cur] > actionScores[prev]) {
+        return cur;
+      }
+      return prev;
+    });
```

Comment:
Have you thought of modifying the `selectAction` method instead of duplicating `evaluateBandit`?

Reply:
This was my first inclination as well however this use fo the building blocks is pretty clean too, and allows for us to do things like move the tie-breaking logic up and not do logging and such so I'm ok with it

Reply:
I did, but it seemed un-necessary to drill down the `selctBestAction` arg that far.